File: _정답코드/2주차/06_할_일_계획하기/solution.cpp
#include <string>
#include <vector>

using namespace std;

bool cmp(vector<int> a, vector<int> b){
    return a[1] < b[1];
}

int solution(vector<vector<int>> tasks) {

    sort(tasks.begin(), tasks.end(), cmp);

    int count = 0;
    int current_time = 0;

    for(auto task : tasks){
        if(current_time <= task[0]){
            current_time = task[1];
            count += 1;
        }
    }
    return count;
}

File: _정답코드/2주차/06_할_일_계획하기/Solution.java
import java.util.*;

public class Solution {
	public int solution(int[][] tasks) {
        List<int[]> tasksList = new ArrayList<>();
        for(int[] task : tasks){
            tasksList.add(task);
        }

        Collections.sort(tasksList, (a, b) -> (a[1] - b[1]));

        int count = 0;
        int current_time = 0;

        for(int[] task : tasksList){
            if(current_time <= task[0]){
                current_time = task[1];
                count += 1;
            }
        }

        return count;
	}
}

